File: Python/algorytmy/liczby23.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  liczby23.py

def liczby2():
    """
    Funkcja drukuje wszystkie liczby 2-cyfrowe, w których cyfry nie powtarzają się. Funkcja zwraca ich liczbę.
    Wykluczone liczby: 11, 22, 33 itd.
    """
    
    ile = 0
    
    for i in range(1, 10):
        for j in range(0, 10):
            if i == j:
                print("", end = '')
            else:
                print("{}{} ".format(i,j), end='')
                ile = ile + 1
    
    return ile
    
def liczby3():
    """
    Funkcja drukuje wszystkie liczby 3-cyfrowe, w których cyfry nie powtarzają się. Funkcja zwraca ich liczbę.
    Wykluczone liczby: 111, 222, 333, 110, 211 itd.
    """
    
    ile = 0 # licznik
    
    for i in range(1, 10): # cyfra setek
        for j in range(0, 10): # cyfra dziesiątek
            for k in range(0, 10): # cyfra jedności
                if i == j or i == k or k == j:
                    print("", end = '')
                else:
                    print("{}{}{} ".format(i,j,k), end='')
                    ile = ile + 1
    
    return ile

def main(args):
    print("Liczby 2-cyfrowych: ", liczby2())
    print("Liczby 3-cyfrowych: ", liczby3())
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/liczby23.cpp
/*
 * liczby23.cpp
 */


#include <iostream>
#include <iomanip>

using namespace std;

int liczby2() {
    int ile = 0;
    for (int i = 1; i < 10 ; i++) {
        for (int j = 0; j < 10; j++) {
            if (i != j) {
                cout << i << j << " "; ile++;}
        }
    }
    return ile;
}

int liczby3() {
    int ile = 0;
    for (int i = 1; i < 10 ; i++) {
        for (int j = 0; j < 10; j++) {
            for (int k = 0; k < 10; k++) {
                if (i != j && i != k && j != k) {
                    cout << i << j << k << " "; 
                    ile++;
                    }
            }
        }
    }
    return ile;
}

int main(int argc, char **argv)
{
	// int ile = liczby2();
    // cout << "\n\nLiczb 2-cyfrowych: " << ile << endl;
	int ile = liczby3();
    cout << "\n\nLiczb 3-cyfrowych: " << ile << endl;
	return 0;
}

